circuitbreaker/circuitbreaker.go

```diff
@@ -10,14 +10,30 @@ import (
 	"github.com/slok/goresilience/metrics"
 )
 
-type state string
+type state uint
 
 const (
-	stateOpen     state = "open"
-	stateHalfOpen state = "halfopen"
-	stateClosed   state = "closed"
+	stateNew   state = iota
```

Comment:
New state is the same as a closed state because a new circuitbreaker starts in a closed state. Do you have a use case where you need to know the difference between new and closed?

Reply:
My choice to implement ```stateNew``` and the transition to ```stateClosed``` on first traverse provided a mechanism to ensure that the state of the breaker was recorded on first use.

Otherwise, I found that there are no state metrics recorded until the first state change occurs. 

I might take a few moments to review how I'm using the circuitbreaker instance by my application.